src/components/common/TranscriptMetadataCard.tsx

```diff
@@ -38,6 +39,11 @@ const TranscriptMetadataComponent = ({
  
   const formattedDate =   isDate(convertedDate) ? format(convertedDate, "d MMMM, yyyy") : "";
 
+  const topicsWithTitles =  topics.map(topic => {
+    const currentTopic =  allTopics.find( topicData => topicData.slug === topic );
+    return currentTopic;
+  } )
+
```

Comment:
Can we handle this at the Contentlayer level? Specifically, replacing topic slugs with their full names during the Contentlayer processing step. This would eliminate the need to perform the replacement every time we display topic tags, such as on the source page, where this replacement is currently not accounted for:
![image](https://github.com/user-attachments/assets/96ccce26-1ba0-4197-9e3d-4834132c44d9)

Reply:
@0tuedon any update on this?
is the suggestion clear?

Reply:
Yh, I am working on putting the logic on contentlayer the file, this will also fix it for our /topics page

Reply:
> Can we handle this at the Contentlayer level? Specifically, replacing topic slugs with their full names during the Contentlayer processing step. This would eliminate the need to perform the replacement every time we display topic tags, such as on the source page, where this replacement is currently not accounted for


But if we replace the topic slugs with their names then we need to to createSlug everytime we want a link?

Reply:
mmm, true.

For clarity, to get the topic name do we unsluggify or we have a lookup with the actual name?
cc: @kouloumos @0tuedon

Reply:
Currently what I did, which @kouloumos pointed out is to find the topics by looping through allTopics and get the name for them. This only fixes it one place at a time.

Reply:
We might need to change it in most places where we do `transcript.tags`. if that's fine I can move on with it

Reply:
> @0tuedon, how was it done for topics page?

``` const topicsWithTitles =  topics.map(topic => {
    const currentTopic =  allTopics.find( topicData => topicData.slug === topic );
    return currentTopic;
  } )
  ```
  
  we are going with a new approach by adding a field to the transcripts called tagsDetailed temporarily.
  
  and I will push this in a few hours then we test to see no link breaks 
  
  and then I will have another commit to directly overwrite the tags.

Reply:
having a new computed field is fine  👍